apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/os/OSUtil.java

```diff
@@ -116,6 +116,9 @@ public static int getProcessNo() {
         }
         List<String> allIPV4 = getAllIPV4();
         if (allIPV4.size() > 0) {
+            if (allIPV4.size() >= ipv4Size) {
+                allIPV4.subList(0, ipv4Size);
```

Comment:
From my reading of the `ArrayList#subList`, the input parameter would not be changed, the new list is the return list. Could you check locally?